R/mod_general_overview_page.R

```diff
@@ -61,51 +61,62 @@ mod_general_overview_page_ui <- function(id) {
 #' 
 #' 
 #' @importFrom future future value
-#' @importFrom hackeRnews get_comments get_top_stories
+#' @importFrom hackeRnews get_comments
 #' 
     
 mod_general_overview_page_server <- function(input, output, session) {
   ns <- session$ns
   
-  top_stories_promise <- reactiveVal()
   comments_promise <- reactiveVal()
   data_fetching_interval_event <- reactiveTimer(DATA_FETCHING_INTERVAL)
   
   output$hovered_node_text <- renderText({
     req(comments_graph$hovered_node())
     req(comments_promise())
     
-    comments_promise() %...>%
-      filter(id == comments_graph$hovered_node()$id) %...>%
-      pull(text) %...>% 
-      remove_html()
+    comments_promise() %...>% {
+        comments <- .
+        validate(need(!is.null(comments), "Unfortunately there are no comments for this story."))
+        comments %>% pull(text)
+      } %...>%
+        filter(id == comments_graph$hovered_node()$id) %...>%
+        pull(text) %...>%
+        remove_html()
   })
   
   observe({
     data_fetching_interval_event()
-    top_stories_promise(future(get_top_stories(max_items = 10)))
   })
   
-  observe({
-    req(stories_table$selected_story())
-    req(top_stories_promise())
-    
-    top_stories <- value(top_stories_promise())
-    selected_top_story <- top_stories[[stories_table$selected_story()]]
-    comments_with_sentiment_promise <- future(get_comments(selected_top_story)) %...>%
-      mutate(sentiment = calculate_sentiment(text) %>% round(digits = 2))
+  clip <- function(x, min_value, max_value) {
+    ifelse(x <= min_value,  min_value, ifelse(x >= max_value, max_value, x))
+  }
+
+  observeEvent(stories_table$selected_story(), {
+    selected_story <- stories_table$selected_story()
+    req(selected_story)
+    comments_with_sentiment_promise <- future(get_comments(selected_story)) %...>% {
+      comments <- .
+      if (!is.null(comments)) {
+        mutate(comments, sentiment = clip(calculate_sentiment(text), -10, 10) %>% round(digits = 2))
+      }
+      # else {
+      #   NULL
+      # }
```

Comment:
Please, don't leave commented code :)

Reply:
yea sorry, was exploring awesomeness of Rlang